Editor/CustomEditors/PaletteEntryDrawer.cs

```diff
@@ -95,8 +95,53 @@ public override void OnGUI(Rect position, SerializedProperty property, PaletteEn
             DrawLabel(position, labelRect, property, label, entry);
         }
 
+        /// <inheritdoc />
+        public override void OnListGUI(Rect position, SerializedProperty property, PaletteEntry baseEntry, bool isSelected)
+        {
+            EntryType entry = (EntryType)baseEntry;
+
+            if (Event.current.type == EventType.MouseDown && Event.current.button == 1 &&
+                position.Contains(Event.current.mousePosition))
+            {
+                ShowContextMenu(entry, property);
+                return;
+            }
+
+            if (Event.current.type != EventType.Repaint)
+                return;
+
+            if (isSelected)
+            {
+                GUIStyle selectionStyle = "RL Element";
+                selectionStyle.Draw(position, false, true, true, true);
+            }
+
+            Texture2D icon = GetListIcon(entry);
+            if (icon == null)
+            {
+                DrawListEntry(position, property, entry);
+            }
+            else
+            {
+                bool isSelectedAndHasFocus =
+                    isSelected && EditorWindow.focusedWindow == PaletteDrawing.ActivePaletteWindow;
+                GUIContent content = new GUIContent(entry.Name, icon);
+                EditorStyles.label.Draw(position, content, false, false, isSelectedAndHasFocus, isSelectedAndHasFocus);
+            }
+        }
+
+        protected virtual Texture2D GetListIcon(EntryType entry)
+        {
+            return null;
+        }
+
         protected abstract void DrawContents(Rect position, SerializedProperty property, EntryType entry);
 
+        protected virtual void DrawListEntry(Rect position, SerializedProperty property, EntryType entry)
+        {
+            EditorGUI.LabelField(position, "No GUI implemented");
```

Comment:
Maybe this should be abstract? Are drawers not required to implement this?

Reply:
There is a default draw implementation. You can however supply your own implementation by overriding this (instead of the `GetListIcon` method) so this doesn't need to be abstract!